File: src/renderer/services/messageService.ts
import { BackendProvider, ChatMessage } from '@shared/types';
import { getBackend, BackendException } from './base';

/**
 * 获取远程会话的消息
 * - Kimi: 调用 getChatDetail，返回会话概要作为消息
 * - 其他后端: 暂不支持
 */
export async function fetchRemoteMessages(
  provider: BackendProvider,
  remoteChatId: string,
  sessionId: string,
  token: string
): Promise<ChatMessage[]> {
  const backend = getBackend(provider);
  backend.setAccessToken(token);

  try {
    switch (provider) {
      case BackendProvider.Kimi: {
        // Kimi API 只返回会话概要，不是完整消息列表
        const detail = await (backend as any).getChatDetail(remoteChatId);
        if (detail && detail.messageContent) {
          return [{
            id: crypto.randomUUID(),
            sessionId,
            role: 'assistant',
            content: detail.messageContent,
            createdAt: Date.now(),
          }];
        }
        return [];
      }
      case BackendProvider.DeepSeek:
      case BackendProvider.Doubao:
      case BackendProvider.Qianwen:
        throw new BackendException(`${provider} 消息获取暂未支持`);
    }
  } catch (error) {
    if (error instanceof BackendException) {
      throw error;
    }
    throw new BackendException(`获取消息失败: ${(error as Error).message}`);
  }
}

/**
 * 从会话ID中提取远程chat ID
 * 格式: kimi_xxxxx -> xxxxx
 */
export function extractRemoteId(sessionId: string): string {
  if (sessionId.includes('_')) {
    return sessionId.substring(sessionId.indexOf('_') + 1);
  }
  return sessionId;
}

File: lib/backend/message_service.dart
import 'package:lingmeng/backend/kimi/kimi.dart';
import 'package:lingmeng/backend/providers.dart';
import 'package:lingmeng/models/chat.dart';
import 'package:lingmeng/app/storage/token_storage.dart';

/// 消息获取服务
class MessageService {
  /// 从指定后端获取会话消息
  Future<List<ChatMessage>> fetchMessages(
    BackendProvider provider,
    String remoteChatId,
    String sessionId,
  ) async {
    final token = TokenStorage.getToken(provider);
    if (token == null || token.isEmpty) {
      throw Exception('${provider.displayName} 未配置 Token');
    }

    switch (provider) {
      case BackendProvider.kimi:
        final kimi = Kimi();
        kimi.setAccessToken(token);
        final detail = await kimi.getChatDetail(chatId: remoteChatId);
        // Kimi GetChat API 只返回会话概要，不包含消息列表
        // 将 messageContent 作为单条消息返回
        if (detail.messageContent.isNotEmpty) {
          return [
            ChatMessage(
              id: '${provider.name}_${detail.id}_msg',
              sessionId: sessionId,
              role: 'assistant',
              content: detail.messageContent,
              createdAt: _parseTime(detail.updateTime),
            ),
          ];
        }
        return [];

      case BackendProvider.deepseek:
      case BackendProvider.doubao:
      case BackendProvider.qianwen:
        // TODO: 实现其他后端的消息获取
        throw UnimplementedError('${provider.displayName} 消息获取暂未实现');
    }
  }

  DateTime _parseTime(String? timeStr) {
    if (timeStr == null || timeStr.isEmpty) {
      return DateTime.now();
    }
    try {
      return DateTime.parse(timeStr);
    } catch (e) {
      return DateTime.now();
    }
  }
}
